src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -128,7 +128,7 @@ func (r *ClusterReconciler) Reconcile(
 	nodeportSvc := resources.NewNodePortService(r.Client, &redpandaCluster, r.Scheme, nodeports, log)
 
 	clusterPorts := []resources.NamedServicePort{}
-	if proxyAPIExternal != nil {
+	if proxyAPIExternal != nil && proxyAPIInternal != nil {
```

Comment:
is this related to this PR? 🤔

Reply:
Not in the PR, that's why I created new commit for it.